components/purchase_modal/purchase_modal.tsx

```diff
@@ -179,7 +193,183 @@ function Card(props: CardProps) {
         trackEvent(TELEMETRY_CATEGORIES.CLOUD_ADMIN, 'click_see_how_billing_works');
         window.open(CloudLinks.BILLING_DOCS, '_blank');
     };
-    return (
+
+    const [usersCount, setUsersCount] = useState(props.usersCount.toString());
+    const [monthlyPrice, setMonthlyPrice] = useState(props.monthlyPrice * props.usersCount);
+    const [yearlyPrice, setYearlyPrice] = useState(props.yearlyPrice * props.usersCount);
+    const [priceDifference, setPriceDifference] = useState((props.monthlyPrice - props.yearlyPrice) * props.usersCount);
+    const [isMonthly, setIsMonthly] = useState(props.isInitialPlanMonthly);
+
+    const monthlyContainerName = 'bottom';
+    const yearlyContainerName = 'bottom bottom-yearly';
+    const initialContainerName = props.isInitialPlanMonthly ? monthlyContainerName : yearlyContainerName;
+    const [containerClassname, setContainerClassname] = useState(initialContainerName);
+
+    const {formatMessage} = props.intl;
+
+    const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const value = event.target.value;
+        setUsersCount(value);
+        const numValue = Number(value);
+        if (numValue && numValue > 0) {
+            setMonthlyPrice(numValue * props.monthlyPrice);
+            setYearlyPrice(numValue * props.yearlyPrice);
+            setPriceDifference((props.monthlyPrice - props.yearlyPrice) * numValue);
+        }
+    };
+
+    const checkValidNumber = () => {
```

Comment:
Since we don't need to allow the user to represent decimals, everything they input should always be a valid state (e.g. don't need to let user type `3.` when they are trying to type `3.5). Instead of setting any input and validating on the next render, can we check that event.target.value is a valid input, and only set it in state if it is? This will also let us skip having to show an error message for entering invalid numbers, and we'll only need to show them for if the user tries to enter less users than they currently have.